File: jump/cp_5_square_0.jl
# NLP written by GAMS Convert at 05/15/24 11:33:45
#
# Equation counts
#     Total        E        G        L        N        X        C        B
#        35        5       20       10        0        0        0        0
#
# Variable counts
#                  x        b        i      s1s      s2s       sc       si
#     Total     cont   binary  integer     sos1     sos2    scont     sint
#        16       16        0        0        0        0        0        0
# FX      0
#
# Nonzero counts
#     Total    const       NL
#       110       50       60
#
# Reformulation has removed 1 variable and 1 equation

using JuMP
using MathOptInterface
const MOI = MathOptInterface

model = m = Model()

@variable(m, -0.5 <= x1 <= 0.5, start=0)
@variable(m, -0.5 <= x2 <= 0.5, start=0)
@variable(m, -0.5 <= x3 <= 0.5, start=0)
@variable(m, -0.5 <= x4 <= 0.5, start=0)
@variable(m, -0.5 <= x5 <= 0.5, start=0)
@variable(m, -0.5 <= x6 <= 0.5, start=0)
@variable(m, -0.5 <= x7 <= 0.5, start=0)
@variable(m, -0.5 <= x8 <= 0.5, start=0)
@variable(m, -0.5 <= x9 <= 0.5, start=0)
@variable(m, -0.5 <= x10 <= 0.5, start=0)
@variable(m, x11, start=0)
@variable(m, x12, start=0)
@variable(m, x13, start=0)
@variable(m, x14, start=0)
@variable(m, x15, start=0)
@variable(m, x16, start=0)

@objective(m, Max, x16)

@NLconstraint(m, e1, (x1 - x2)^2 + (x6 - x7)^2 - (x11 + x12)^2 >= 0)
@NLconstraint(m, e2, (x1 - x3)^2 + (x6 - x8)^2 - (x11 + x13)^2 >= 0)
@NLconstraint(m, e3, (x1 - x4)^2 + (x6 - x9)^2 - (x11 + x14)^2 >= 0)
@NLconstraint(m, e4, (x1 - x5)^2 + (x6 - x10)^2 - (x11 + x15)^2 >= 0)
@NLconstraint(m, e5, (x2 - x3)^2 + (x7 - x8)^2 - (x12 + x13)^2 >= 0)
@NLconstraint(m, e6, (x2 - x4)^2 + (x7 - x9)^2 - (x12 + x14)^2 >= 0)
@NLconstraint(m, e7, (x2 - x5)^2 + (x7 - x10)^2 - (x12 + x15)^2 >= 0)
@NLconstraint(m, e8, (x3 - x4)^2 + (x8 - x9)^2 - (x13 + x14)^2 >= 0)
@NLconstraint(m, e9, (x3 - x5)^2 + (x8 - x10)^2 - (x13 + x15)^2 >= 0)
@NLconstraint(m, e10, (x4 - x5)^2 + (x9 - x10)^2 - (x14 + x15)^2 >= 0)
@constraint(m, e11, x11 - x16 == 0)
@constraint(m, e12, x12 - x16 == 0)
@constraint(m, e13, x13 - x16 == 0)
@constraint(m, e14, x14 - x16 == 0)
@constraint(m, e15, x15 - x16 == 0)
@constraint(m, e16, x1 + x11 <= 0.5)
@constraint(m, e17, x2 + x12 <= 0.5)
@constraint(m, e18, x3 + x13 <= 0.5)
@constraint(m, e19, x4 + x14 <= 0.5)
@constraint(m, e20, x5 + x15 <= 0.5)
@constraint(m, e21, x1 - x11 >= -0.5)
@constraint(m, e22, x2 - x12 >= -0.5)
@constraint(m, e23, x3 - x13 >= -0.5)
@constraint(m, e24, x4 - x14 >= -0.5)
@constraint(m, e25, x5 - x15 >= -0.5)
@constraint(m, e26, x6 + x11 <= 0.5)
@constraint(m, e27, x7 + x12 <= 0.5)
@constraint(m, e28, x8 + x13 <= 0.5)
@constraint(m, e29, x9 + x14 <= 0.5)
@constraint(m, e30, x10 + x15 <= 0.5)
@constraint(m, e31, x6 - x11 >= -0.5)
@constraint(m, e32, x7 - x12 >= -0.5)
@constraint(m, e33, x8 - x13 >= -0.5)
@constraint(m, e34, x9 - x14 >= -0.5)
@constraint(m, e35, x10 - x15 >= -0.5)

File: pyomo/cp_5_square_0.py
# NLP written by GAMS Convert at 05/15/24 11:46:47
#
# Equation counts
#     Total        E        G        L        N        X        C        B
#        35        5       20       10        0        0        0        0
#
# Variable counts
#                  x        b        i      s1s      s2s       sc       si
#     Total     cont   binary  integer     sos1     sos2    scont     sint
#        16       16        0        0        0        0        0        0
# FX      0
#
# Nonzero counts
#     Total    const       NL
#       110       50       60
#
# Reformulation has removed 1 variable and 1 equation

from pyomo.environ import *

model = m = ConcreteModel()

m.x1 = Var(within=Reals, bounds=(-0.5,0.5), initialize=0)
m.x2 = Var(within=Reals, bounds=(-0.5,0.5), initialize=0)
m.x3 = Var(within=Reals, bounds=(-0.5,0.5), initialize=0)
m.x4 = Var(within=Reals, bounds=(-0.5,0.5), initialize=0)
m.x5 = Var(within=Reals, bounds=(-0.5,0.5), initialize=0)
m.x6 = Var(within=Reals, bounds=(-0.5,0.5), initialize=0)
m.x7 = Var(within=Reals, bounds=(-0.5,0.5), initialize=0)
m.x8 = Var(within=Reals, bounds=(-0.5,0.5), initialize=0)
m.x9 = Var(within=Reals, bounds=(-0.5,0.5), initialize=0)
m.x10 = Var(within=Reals, bounds=(-0.5,0.5), initialize=0)
m.x11 = Var(within=Reals, bounds=(None,None), initialize=0)
m.x12 = Var(within=Reals, bounds=(None,None), initialize=0)
m.x13 = Var(within=Reals, bounds=(None,None), initialize=0)
m.x14 = Var(within=Reals, bounds=(None,None), initialize=0)
m.x15 = Var(within=Reals, bounds=(None,None), initialize=0)
m.x16 = Var(within=Reals, bounds=(None,None), initialize=0)

m.obj = Objective(sense=maximize, expr= m.x16)

m.e1 = Constraint(expr= (m.x1 - m.x2)**2 + (m.x6 - m.x7)**2 - (m.x11 + m.x12)**
    2 >= 0)
m.e2 = Constraint(expr= (m.x1 - m.x3)**2 + (m.x6 - m.x8)**2 - (m.x11 + m.x13)**
    2 >= 0)
m.e3 = Constraint(expr= (m.x1 - m.x4)**2 + (m.x6 - m.x9)**2 - (m.x11 + m.x14)**
    2 >= 0)
m.e4 = Constraint(expr= (m.x1 - m.x5)**2 + (m.x6 - m.x10)**2 - (m.x11 + m.x15)
    **2 >= 0)
m.e5 = Constraint(expr= (m.x2 - m.x3)**2 + (m.x7 - m.x8)**2 - (m.x12 + m.x13)**
    2 >= 0)
m.e6 = Constraint(expr= (m.x2 - m.x4)**2 + (m.x7 - m.x9)**2 - (m.x12 + m.x14)**
    2 >= 0)
m.e7 = Constraint(expr= (m.x2 - m.x5)**2 + (m.x7 - m.x10)**2 - (m.x12 + m.x15)
    **2 >= 0)
m.e8 = Constraint(expr= (m.x3 - m.x4)**2 + (m.x8 - m.x9)**2 - (m.x13 + m.x14)**
    2 >= 0)
m.e9 = Constraint(expr= (m.x3 - m.x5)**2 + (m.x8 - m.x10)**2 - (m.x13 + m.x15)
    **2 >= 0)
m.e10 = Constraint(expr= (m.x4 - m.x5)**2 + (m.x9 - m.x10)**2 - (m.x14 + m.x15)
    **2 >= 0)
m.e11 = Constraint(expr= m.x11 - m.x16 == 0)
m.e12 = Constraint(expr= m.x12 - m.x16 == 0)
m.e13 = Constraint(expr= m.x13 - m.x16 == 0)
m.e14 = Constraint(expr= m.x14 - m.x16 == 0)
m.e15 = Constraint(expr= m.x15 - m.x16 == 0)
m.e16 = Constraint(expr= m.x1 + m.x11 <= 0.5)
m.e17 = Constraint(expr= m.x2 + m.x12 <= 0.5)
m.e18 = Constraint(expr= m.x3 + m.x13 <= 0.5)
m.e19 = Constraint(expr= m.x4 + m.x14 <= 0.5)
m.e20 = Constraint(expr= m.x5 + m.x15 <= 0.5)
m.e21 = Constraint(expr= m.x1 - m.x11 >= -0.5)
m.e22 = Constraint(expr= m.x2 - m.x12 >= -0.5)
m.e23 = Constraint(expr= m.x3 - m.x13 >= -0.5)
m.e24 = Constraint(expr= m.x4 - m.x14 >= -0.5)
m.e25 = Constraint(expr= m.x5 - m.x15 >= -0.5)
m.e26 = Constraint(expr= m.x6 + m.x11 <= 0.5)
m.e27 = Constraint(expr= m.x7 + m.x12 <= 0.5)
m.e28 = Constraint(expr= m.x8 + m.x13 <= 0.5)
m.e29 = Constraint(expr= m.x9 + m.x14 <= 0.5)
m.e30 = Constraint(expr= m.x10 + m.x15 <= 0.5)
m.e31 = Constraint(expr= m.x6 - m.x11 >= -0.5)
m.e32 = Constraint(expr= m.x7 - m.x12 >= -0.5)
m.e33 = Constraint(expr= m.x8 - m.x13 >= -0.5)
m.e34 = Constraint(expr= m.x9 - m.x14 >= -0.5)
m.e35 = Constraint(expr= m.x10 - m.x15 >= -0.5)
